integrated_channels/moodle/client.py

```diff
@@ -150,6 +151,100 @@ def _get_access_token(self):
                 response.status_code
             )
 
+    @moodle_request_wrapper
+    def _get_enrolled_users(self, course_id):
+        """
+        placeholder
+        """
+        params = {
+            'wstoken': self.token,
+            'wsfunction': 'core_enrol_get_enrolled_users',
+            'courseid': course_id,
+            'moodlewsrestformat': 'json'
+        }
+        user_response = requests.get(
+            urljoin(
+                self.enterprise_configuration.moodle_base_url,
+                self.MOODLE_API_PATH,
+            ),
+            params=params
+        )
+        return user_response
+
+    def get_creds_of_user_in_course(self, course_id, user_email):
+        """
+        placeholder
+        """
+        response = self._get_enrolled_users(course_id)
+        parsed_response = response.json()
+        user_id = None
+
+        if isinstance(parsed_response) == list:
+            for enrollment in parsed_response:
+                if enrollment.get('email') == user_email:
+                    user_id = enrollment.get('id')
+                    break
+        if not user_id:
+            raise ClientError(
+                "MoodleAPIClient request failed: 404 User enrollment not found under user={} in course={}.".format(
+                    user_email,
+                    course_id
+                ),
+                HTTPStatus.NOT_FOUND.value
+            )
+        return user_id
+
+    @moodle_request_wrapper
+    def _get_course_contents(self, course_id):
+        """
+        placeholder
+        """
+        params = {
+            'wstoken': self.token,
+            'wsfunction': 'core_course_get_contents',
+            'courseid': course_id,
+            'moodlewsrestformat': 'json'
+        }
+        response = requests.get(
+            urljoin(
+                self.enterprise_configuration.moodle_base_url,
+                self.MOODLE_API_PATH,
+            ),
+            params=params
+        )
+        return response
+
+    def get_course_final_grade_module(self, course_id):
+        """
+        placeholder
+        """
+        response = self._get_course_contents(course_id)
+
+        course_module_id = None
+        if isinstance(response.json()) is list:
+            for course in response.json():
+
+                # This is one point where I think we can make this configurable-
+                # A freshly made course will have "General" module that this code
+                # is assuming the `final grade` activity will live. We could pretty easily
+                # have the user give us the ID of the module where the activity lives and
+                # even default to the 'General'
+                if course.get('name') == 'General':
+                    modules = course.get('modules')
+                    for module in modules:
+                        if module.get('name') == '(edX integration) Final Grade':
```

Comment:
is this a constant somewhere, or should be if used outside of this code as well?

Reply:
ah yeah I would constantify it esp since it's used again later here too

Reply:
It is not a constant, that's a great point and one that I meant to do but forgot. Good catch, thanks!!